satpy/readers/file_handlers.py

```diff
@@ -122,15 +122,112 @@ def sensor_names(self):
         """List of sensors represented in this file."""
         raise NotImplementedError
 
-    def available_datasets(self):
-        """Get information of available datasets in file.
+    def file_type_matches(self, ds_ftype):
+        """This file handler's type can handle this dataset's file type.
 
-        This is used for dynamically specifying what datasets are available
-        from a file instead of those listed in a YAML configuration file.
+        Args:
+            ds_ftype (str or list): File type or list of file types that a
+                dataset is configured to be loaded from.
 
-        Returns: Iterator of (DatasetID, dict) pairs where dict is the
-                 dataset's metadata, similar to that specified in the YAML
-                 configuration files.
+        Returns: ``True`` if this file handler object's type matches the
+            dataset's file type(s), ``False`` otherwise.
 
         """
-        raise NotImplementedError
+        if isinstance(ds_ftype, str) and ds_ftype == self.filetype_info['file_type']:
+            return True
+        elif self.filetype_info['file_type'] in ds_ftype:
+            return True
+        return None
+
+    def available_datasets(self, configured_datasets=None):
+        """Get information of available datasets in this file.
+
+        This is used for dynamically specifying what datasets are available
+        from a file in addition to what's configured in a YAML configuration
+        file. Note that this method will only be called once for each
+        "file type"; the first file handler for each type.
+
+        This method should **not** update values of the dataset information
+        dictionary **unless* this file handler has a matching file type
+        (the data could be loaded from this object in the future) and at least
+        **one** :class:`satpy.dataset.DatasetID` key is also modified.
+        Otherwise, this file type may override the information provided by
+        a more preferred file type (as specified in the YAML file).
+        It is recommended that any non-ID metadata be updated during the
+        :meth:`BaseFileHandler.get_dataset` part of loading.
+        This method is not guaranteed that it will be called before any
+        other file type's handler.
+        The availability "boolean" not being ``None`` does not mean that a
+        file handler called later can't provide an additional dataset, but
+        it must provide more identifying (DatasetID) information to do so
+        and should yield its new dataset in addition to the previous one.
+
+        Args:
+            configured_datasets (list): Series of (bool, dict) in the same
```

Comment:
could it be mentioned already here that the fist element can be None ? eg `(bool or None, dict)`